dotCMS/src/main/java/com/dotcms/saml/DotSamlProxyFactory.java

```diff
@@ -0,0 +1,239 @@
+import com.liferay.portal.model.User;
+import com.liferay.util.FileUtil;
+import io.vavr.control.Try;
+import org.apache.felix.framework.OSGIUtil;
+import org.tuckey.web.filters.urlrewrite.NormalRule;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * This is the proxy to provides the object to interact with the Saml Osgi Bundle
+ *
+ * @author jsanca
+ */
+public class DotSamlProxyFactory {
+
+    public static final String SAML_APP_CONFIG_KEY = "dotsaml-config";
+    public static final String PROPERTIES_PATH = File.separator + "saml" + File.separator + "dotcms-saml-default.properties";
+
+    private static final String ASSETS_PATH = Config.getStringProperty("ASSET_REAL_PATH",
+            FileUtil.getRealPath(Config.getStringProperty("ASSET_PATH", "/assets")));
+    private static final String IDP_FILE_PATH = ASSETS_PATH + PROPERTIES_PATH;
+
+    private final MessageObserver    messageObserver    = new DotLoggerMessageObserver();
+    private final AppsAPI            appsAPI            = APILocator.getAppsAPI();
+    private final IdentityProviderConfigurationFactory identityProviderConfigurationFactory =
+            new DotIdentityProviderConfigurationFactoryImpl(this.appsAPI, APILocator.getHostAPI());
+
+    private SamlServiceBuilder        samlServiceBuilder         = null;
+    private SamlConfigurationService  samlConfigurationService   = null;
+    private SamlAuthenticationService samlAuthenticationService  = null;
+
+    private static class SingletonHolder {
+
+        private static final DotSamlProxyFactory INSTANCE = new DotSamlProxyFactory();
+    }
+    /**
+     * Get the instance.
+     * @return DotSamlFactory
+     */
+    public static DotSamlProxyFactory getInstance() {
+
+        addRedirects();
+        return DotSamlProxyFactory.SingletonHolder.INSTANCE;
+    } // getInstance.
+
+    /**
+     * Returns the dotCMS implementation of the identity provider config.
+     * This one basically returns (if exists) the configuration for a idp for a host
+     * @return IdentityProviderConfigurationFactory
+     */
+    public IdentityProviderConfigurationFactory identityProviderConfigurationFactory() {
+
+        return identityProviderConfigurationFactory;
+    }
+
+    private static void addRedirects() {
+
+        final NormalRule rule = new NormalRule();
+        rule.setFrom("^\\/dotsaml\\/("+String.join("|", DotSamlResource.dotsamlPathSegments)+")\\/(.+)$");
+        rule.setToType("forward");
+        rule.setTo("/api/v1/dotsaml/$1/$2");
+        rule.setName("Dotsaml REST Service Redirect");
+        DotUrlRewriteFilter urlRewriteFilter = DotUrlRewriteFilter.getUrlRewriteFilter();
+        try {
+            if(urlRewriteFilter != null) {
+                urlRewriteFilter.addRule(rule);
+            }else {
+                throw new Exception();
+            }
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)